File: realworld-back/realworld-service/src/main/java/com/realworld/controller/TagController.java
package com.realworld.controller;

import com.realworld.result.Result;
import com.realworld.service.TagsService;
import com.realworld.vo.TagsVO;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@Tag(name = "标签管理")
@RequestMapping("/tags")
@RestController
public class TagController {

	@Autowired
	private TagsService tagsService;

	@Operation(summary = "获取标签列表")
	@GetMapping
	public Result<List<TagsVO> > getTags() {
		return Result.success(tagsService.listTag());
	}
}

File: backend-springboot/src/main/kotlin/com/yyjy/controller/TagController.kt
package com.yyjy.controller

import com.yyjy.common.ApiRes
import com.yyjy.service.TagService
import io.swagger.v3.oas.annotations.tags.Tag
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@Tag(name = "标签模块")
@RequestMapping("/tags")
@RestController
class TagController(
    private val tagService: TagService
) {
    @GetMapping
    fun getAllTags(): ApiRes<List<String>> {
        return ApiRes.ok(tagService.getAllTags())
    }
}